uint8 agent_id
uint8[] roles
uint8 gamestate
